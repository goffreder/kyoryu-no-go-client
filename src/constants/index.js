export const ui = {
    GRID_SIZE: 40,
};

export const board = {
    EMPTY: 0,
    BLACK: 1,
    WHITE: 2,
};

export const msg = {
    ATARI: 'ATARI!',
    SUICIDE: 'SUICIDE!',
    GAME_OVER: 'GAME OVER!',
};
