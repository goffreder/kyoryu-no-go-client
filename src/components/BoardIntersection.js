import React, { Component } from 'react';
import { func, number, oneOf, bool, arrayOf } from 'prop-types';
import classNames from 'classnames';

import Stone from './Stone';
import { board } from '../constants';

export default class extends Component {
    static propTypes = {
        row: number.isRequired,
        col: number.isRequired,
        color: oneOf([board.EMPTY, board.BLACK, board.WHITE]),
        play: func,
        width: number,
        isTopEdge: bool,
        isRightEdge: bool,
        isBottomEdge: bool,
        isLeftEdge: bool,
        isStarPoint: bool,
        readonly: bool,
        lastMove: arrayOf(number),
    };

    static defaultProps = {
        color: board.EMPTY,
        width: 100,
        readonly: false,
        lastMove: null,
    };

    handleClick = () => {
        !this.props.readonly &&
            this.props.play &&
            this.props.play(this.props.row, this.props.col);
    };

    render() {
        const isLastMove = this.props.lastMove
            ? this.props.lastMove[0] === this.props.row &&
              this.props.lastMove[1] === this.props.col
            : false;

        return (
            <div
                style={{
                    float: 'left',
                    lineHeight: '1',
                    paddingTop: this.props.width + '%',
                    position: 'relative',
                    textAlign: 'center',
                    width: this.props.width + '%',
                }}
                className={classNames(
                    'intersection',
                    this.props.isStarPoint ? 'hoshi' : null,
                )}
                onClick={this.handleClick}
            >
                <div
                    style={{
                        cursor:
                            this.props.color !== board.EMPTY ||
                            this.props.readonly
                                ? 'auto'
                                : 'pointer',
                        position: 'absolute',
                        top: 0,
                        right: 0,
                        bottom: 0,
                        left: 0,
                    }}
                >
                    {this.props.isStarPoint ? (
                        <div
                            style={{
                                background: '#63380E',
                                borderRadius: 9000,
                                height: '20%',
                                position: 'absolute',
                                left: '50%',
                                top: '50%',
                                transform: 'translate(-50%, -50%)',
                                width: '20%',
                                minWidth: 4,
                                minHeight: 4,
                            }}
                        />
                    ) : null}
                    <div
                        style={{
                            background: '#63380E',
                            height: '1px',
                            position: 'absolute',
                            top: '50%',
                            right: this.props.isRightEdge ? '50%' : 0,
                            left: this.props.isLeftEdge ? '50%' : 0,
                            transform: 'translateY(-50%)',
                        }}
                    />
                    <div
                        style={{
                            background: '#63380E',
                            position: 'absolute',
                            left: '50%',
                            top: this.props.isTopEdge ? '50%' : 0,
                            bottom: this.props.isBottomEdge ? '50%' : 0,
                            transform: 'translateX(-50%)',
                            width: '1px',
                        }}
                    />
                    {this.props.color !== board.EMPTY ? (
                        <Stone
                            color={this.props.color}
                            isLastMove={isLastMove}
                        />
                    ) : null}
                </div>
            </div>
        );
    }
}
