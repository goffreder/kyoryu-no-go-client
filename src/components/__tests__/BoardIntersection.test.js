import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';
import BoardIntersection from '../BoardIntersection';
import { board } from '../../constants';

describe('BoardIntersection component', () => {
    it('displays an empty intersection', () => {
        const { container } = render(
            <BoardIntersection row={0} col={0} play={() => {}} />,
        );

        expect(container.querySelector('div.intersection')).toBeInTheDocument();
    });

    it('displays a star point intersection', () => {
        expect(
            renderer
                .create(
                    <BoardIntersection
                        row={0}
                        col={0}
                        play={() => {}}
                        isStarPoint
                    />,
                )
                .toJSON(),
        ).toMatchSnapshot();
    });

    it('displays edges intersections', () => {
        expect(
            renderer
                .create(
                    <BoardIntersection
                        row={0}
                        col={0}
                        play={() => {}}
                        isTopEdge
                    />,
                )
                .toJSON(),
        ).toMatchSnapshot();
        expect(
            renderer
                .create(
                    <BoardIntersection
                        row={0}
                        col={0}
                        play={() => {}}
                        isLeftEdge
                    />,
                )
                .toJSON(),
        ).toMatchSnapshot();
        expect(
            renderer
                .create(
                    <BoardIntersection
                        row={0}
                        col={0}
                        play={() => {}}
                        isRightEdge
                    />,
                )
                .toJSON(),
        ).toMatchSnapshot();
        expect(
            renderer
                .create(
                    <BoardIntersection
                        row={0}
                        col={0}
                        play={() => {}}
                        isBottomEdge
                    />,
                )
                .toJSON(),
        ).toMatchSnapshot();
    });

    it('displays a black stone on an intersection', () => {
        const { container } = render(
            <BoardIntersection
                row={0}
                col={0}
                play={() => {}}
                color={board.BLACK}
            />,
        );

        expect(
            container.querySelector('.intersection .stone.stone-black'),
        ).toBeInTheDocument();
    });

    it('displays a white stone on an intersection', () => {
        const { container } = render(
            <BoardIntersection
                row={0}
                col={0}
                play={() => {}}
                color={board.WHITE}
            />,
        );

        expect(
            container.querySelector('.intersection .stone.stone-white'),
        ).toBeInTheDocument();
    });

    it('calls "play" prop on intersection click', () => {
        const play = jest.fn();
        const { container } = render(
            <BoardIntersection row={0} col={0} play={play} />,
        );

        fireEvent.click(container.querySelector('.intersection'));
        expect(play).toHaveBeenCalledTimes(1);
    });

    it('should not call "play" prop on readonly intersection click', () => {
        const play = jest.fn();
        const { container } = render(
            <BoardIntersection row={0} col={0} play={play} readonly />,
        );

        fireEvent.click(container.querySelector('.intersection'));
        expect(play).not.toHaveBeenCalled();
    });
});
