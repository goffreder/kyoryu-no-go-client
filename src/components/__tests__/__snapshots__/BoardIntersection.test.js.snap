// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardIntersection component displays a star point intersection 1`] = `
<div
  className="intersection hoshi"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "borderRadius": 9000,
          "height": "20%",
          "left": "50%",
          "minHeight": 4,
          "minWidth": 4,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "20%",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": 0,
          "left": "50%",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
  </div>
</div>
`;

exports[`BoardIntersection component displays edges intersections 1`] = `
<div
  className="intersection"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": 0,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
  </div>
</div>
`;

exports[`BoardIntersection component displays edges intersections 2`] = `
<div
  className="intersection"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": "50%",
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": 0,
          "left": "50%",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
  </div>
</div>
`;

exports[`BoardIntersection component displays edges intersections 3`] = `
<div
  className="intersection"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": 0,
          "position": "absolute",
          "right": "50%",
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": 0,
          "left": "50%",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
  </div>
</div>
`;

exports[`BoardIntersection component displays edges intersections 4`] = `
<div
  className="intersection"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": "50%",
          "left": "50%",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
  </div>
</div>
`;

exports[`BoardIntersection component displays the last placed white stone on an intersection 1`] = `
<div
  className="intersection"
  onClick={[Function]}
  style={
    Object {
      "float": "left",
      "lineHeight": "1",
      "paddingTop": "100%",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "cursor": "auto",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "background": "#63380E",
          "height": "1px",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#63380E",
          "bottom": 0,
          "left": "50%",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "1px",
        }
      }
    />
    <div
      className="stone stone-white"
      style={
        Object {
          "background": "#EEEEF0",
          "borderRadius": 9001,
          "bottom": "5%",
          "boxShadow": Array [
            "inset 0 -3.3333333333333335px 6.666666666666667px 0 rgba(0, 0, 0, .16)",
            "0 1.6666666666666667px 3.3333333333333335px 0 rgba(0, 0, 0, .16)",
            "0 1.6666666666666667px 5px 0 rgba(0, 0, 0, .12)",
          ],
          "left": "5%",
          "position": "absolute",
          "right": "5%",
          "top": "5%",
        }
      }
    >
      <div
        style={
          Object {
            "border": "0.2em solid #101015",
            "borderRadius": "100%",
            "height": "40%",
            "left": "30%",
            "position": "relative",
            "top": "30%",
            "width": "40%",
          }
        }
      />
    </div>
  </div>
</div>
`;
